src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -4278,8 +4219,12 @@ address MacroAssembler::load_and_call(Address entry) {
     assert_alignment(call_pc);
   }
 #endif
+
   relocate(entry.rspec(), [&] {
-    load_link_jump(target, t1);
+    int64_t distance = 0;
+    auipc(tmp, (int32_t)distance + 0x800);
+    ld(tmp, Address(tmp, ((int32_t)distance << 20) >> 20));
+    jalr(tmp);
```

Comment:
Seems we can totally remove the usage of `distance`? could be something like below:
```
auipc(tmp, 0);
ld(tmp, Address(tmp, 0));
jalr(tmp);
```

Reply:
If the above simplication make sense, then what we're doing here is to load from pc as the destination addr, then jump to the addr? I could be misunderstanding.

Reply:
Thanks. I removed `distance` and simplified the code as suggested. Note that the use of PC as address here is only temporary. The relocation created while emitting the stub will ensure this call sequence is subsequently patched to call the stub. I have added code comments for to help understand.

Reply:
Thanks for updating and explanation.